src/twisted/conch/scripts/ckeygen.py

```diff
@@ -339,23 +368,23 @@ def _saveKey(key, options):
 
     comment = f"{getpass.getuser()}@{socket.gethostname()}"
 
-    filepath.FilePath(options["filename"]).setContent(
+    filepath.FilePath(filename).setContent(
         key.toString(
             "openssh",
             subtype=options["private-key-subtype"],
             passphrase=options["pass"],
         )
     )
-    os.chmod(options["filename"], 33152)
+    os.chmod(filename, 33152)
```

Comment:
Also this magic number is terrible :)

```suggestion
    fp.chmod(0o100600)
```

Reply:
Also we should file a bug here, there's arguably a umask race in this code, we should not be chmod'ing things after the fact

Reply:
Definitely makes sense, updated.